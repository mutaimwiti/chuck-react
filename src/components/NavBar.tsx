import React from 'react';
import { Navbar } from 'react-bootstrap';

type NavBarProps = {
  onGoHome(): void;
};

const NavBar: React.FC<NavBarProps> = ({ onGoHome }) => {
  return <Navbar bg="primary" variant="dark">
      <Navbar.Brand href="#" onClick={() => onGoHome()}>
        Chuck React
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
    </Navbar>
  );
}

export default NavBar
