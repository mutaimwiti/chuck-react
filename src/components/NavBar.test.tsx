import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import NavBar, { NavBarProps } from './NavBar';

describe('NavBar', function () {
  let props: NavBarProps;

  beforeEach(() => {
    props = {
      onGoHome: jest.fn(),
    };
  });

  it('should not blow up', () => {
    expect(() => render(<NavBar {...props} />)).not.toThrow();
  });

  it('should match snapshot', () => {
    const { container } = render(<NavBar {...props} />);

    expect(container).toMatchSnapshot();
  });

  describe('when user clicks home', () => {
    it('should invoke onGoHome', () => {
      const { getByText } = render(<NavBar {...props} />);

      expect(props.onGoHome).not.toHaveBeenCalled();

      fireEvent.click(getByText('Chuck React'));

      expect(props.onGoHome).toHaveBeenCalledTimes(1);
    });
  });
});
