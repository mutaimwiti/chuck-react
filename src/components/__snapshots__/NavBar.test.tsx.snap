// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-primary"
  >
    <a
      class="navbar-brand"
      href="#"
    >
      Chuck React
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
  </nav>
</div>
`;
